### ISSUE TITLE
add "references" support

### ISSUE THREAD

**Issue by kpwebb** [sharedstreets/sharedstreets-pbf#2]
Implementation per #1 looks great! Just adding a request to include "references" pbf support to lib cc @DenisCarriere

**Comment by DenisCarriere** [comment]
👍 sounds good

**Comment by DenisCarriere** [comment]
Added References & Metadata to the `sharedstreets-pbf` JS library.

## Reference

```js
const buffer = fs.readFileSync('z-x-y.reference.pbf')

const references = sharedstreetsPbf.reference(buffer)
references[0].id // => 'NxPFkg4CrzHeFhwV7Uiq7K'
```

**Test outputs:**

```json
[
	{
		"id": "Bg1tCb7pjpb9Z8RZVGqBtK",
		"geometryId": "KhqRu35xJWF37XSaxECAmz",
		"formOfWay": "MultipleCarriageway",
		"locationReferences": [
			{
				"intersectionId": "6EyDQdzmBPNdYB8MM2YS19",
				"lat": 40.72031021118164,
				"lon": -74.01203918457031,
				"outboundBearing": 9,
				"distanceToNextRef": 15003
			},
			{
				"intersectionId": "VeMtwF5u5MVn8DM6hiJwGM",
				"lat": 40.72164535522461,
				"lon": -74.01176452636719,
				"inboundBearing": 9
			}
		]
	},
	{
		"id": "EVKrpZExMiG7XX8dX6Q5bF",
		"geometryId": "3myL2RGzdoa5RxVj2mit7V",
		"formOfWay": "SingleCarriageway",
		"locationReferences": [
			{
				"intersectionId": "J5HdLZL3DQMjbLTvdqGWQR",
				"lat": 40.734092712402344,
				"lon": -74.00849151611328,
				"outboundBearing": 258,
				"distanceToNextRef": 13828
			},
			{
				"intersectionId": "AvKv8jN9C8HnR74wFYzkqx",
				"lat": 40.73386764526367,
				"lon": -74.01009368896484,
				"inboundBearing": 267
			}
		]
	}
...
]
```

## Metadata

```js
const buffer = fs.readFileSync('z-x-y.metadata.pbf')

const metadatas = sharedstreetsPbf.metadata(buffer)
metadatas[0].geometryID // => 'HGdvAjtekfDrLFzPevNtf3'
```

**Test outputs:**

```json
[
	{
		"geometryID": "HGdvAjtekfDrLFzPevNtf3",
		"osmMetadata": {
			"waySections": [
				{
					"wayId": "118714476",
					"roadClass": "Trunk",
					"oneWay": true,
					"nodeIds": [
						"42434827",
						"42438684"
					]
				}
			]
		}
	},
	{
		"geometryID": "HjvqyidQRyDCa4EhA5H16D",
		"osmMetadata": {
			"waySections": [
				{
					"wayId": "497191099",
					"roadClass": "Secondary",
					"oneWay": true,
					"nodeIds": [
						"42432116",
						"42436353"
					]
				}
			]
		}
	}
...
]
```

**Comment by DenisCarriere** [comment]
@kpwebb Noticed `geometryId` in Reference is different than Metadata via the lowercase "d".

- geometryID => Metadata
- geometryId => Reference

**Comment by DenisCarriere** [comment]
This change is published under `v0.3.0`
